reproducibility_tests/latest_comparable_paths.jl

```diff
@@ -0,0 +1,110 @@
+        ref_counter_file = joinpath(path, "ref_counter.jl")
+        !isfile(ref_counter_file) && continue
+        ref_counters_in_path[i] = parse(Int, first(readlines(ref_counter_file)))
+    end
+    return ref_counters_in_path
+end
+
+"""
+    latest_comparable_paths(n::Integer)
+
+Returns a vector of strings, containing the `n`
+latest comparable paths based on
+`reproducibility_tests/ref_counter.jl`.
+"""
+function latest_comparable_paths(n = 5)
+    if get(ENV, "BUILDKITE_PIPELINE_SLUG", nothing) != "climaatmos-ci"
+        @warn "Not using climaatmos-ci pipeline slug, assuming no comparable references"
+        @info "Please review output results before merging."
+        return String[]
+    end
+
+    # Note: cluster_data_prefix is also defined in move_output.jl
+    cluster_data_prefix = "/central/scratch/esm/slurm-buildkite/climaatmos-main"
+    # Get (sorted) array of paths, `pop!(sorted_paths)`
+    # is the most recent merged folder.
+    sorted_paths = sorted_dataset_folder(; dir = cluster_data_prefix)
+    if isempty(sorted_paths)
+        @warn "No paths on main found, assuming no comparable references"
+        @info "Please review output results before merging."
+        return String[]
+    end
+    # Find oldest path in main with the same reference
+    # counter as the one in the PR. If none exists,
+    # then assume no comparable references.
+
+    ref_counter_file_PR = joinpath(@__DIR__, "ref_counter.jl")
+    @assert isfile(ref_counter_file_PR)
+    ref_counter_PR = parse(Int, first(readlines(ref_counter_file_PR)))
+
+    ref_counters_main = ref_counters_per_path(sorted_paths)
+    i_comparable_references = findall(ref_counters_main) do ref_counter_main
+        ref_counter_main == ref_counter_PR
+    end
+    if i_comparable_references == nothing
+        @warn "`ref_counter.jl` not found on main, assuming no comparable references"
+        @info "Please review output results before merging."
+        return String[]
+    end
+    @info "Found $(length(i_comparable_references)) comparable references:$i_comparable_references"
+    # Largest ref-counter reference path:
+    paths = map(i -> sorted_paths[i], i_comparable_references)
+    @info "$(length(paths)) paths found:"
+    for p in paths
+        @info "     $p, $(Dates.unix2datetime(stat(p).mtime))"
+    end
+    ref_counter_files_main = map(p -> joinpath(p, "ref_counter.jl"), paths)
+    @info "$(length(ref_counter_files_main)) reference counter paths on central"
+    filter!(isfile, ref_counter_files_main)
+    @info "$(length(ref_counter_files_main)) reference counter paths on central after filtering isfile"
+
+    # for p in paths
+    #     @info "Files in $p:" # for debugging
+    #     for file_on_main in readdir(p)
+    #         @info "   File:`$file_on_main`"
+    #     end
+    # end
+    @assert all(isfile, ref_counter_files_main)
+    ref_counters_main =
+        map(f -> parse(Int, first(readlines(f))), ref_counter_files_main)
```

Comment:
I suggest we add a function `read_ref_coutner` because we do this operation often now.